text/0201-scope-attributes.md

```diff
@@ -0,0 +1,138 @@
+# Introduce Scope Attributes
+
+This OTEP adds attributes to the Scope.
+
+## Motivation
+
+There are a few reasons why adding Scope attributes is a good idea:
+
+- There are 2 known use cases where Scope attributes can solve specific problems:
+  - Add support for [Meter "short_name"](https://github.com/open-telemetry/opentelemetry-specification/pull/2422),
+    represented as an attribute of Meter's Scope.
+  - Add support for differentiating the type of data emitted from the scopes that belong
+    to different data domains, e.g. profiling data emitted as log records or client-side
+    data emitted as log records needs to be differentiated so that it can be easily
+    routed and processed differently in the backends. We don't have a good way to handle
+    this today. The type of the data can be recorded as an attribute Logger's Scope.
```

Comment:
Shouldn't the domain identifying attribute be at the resource level, since it will be repeated for each instrumentation scope anyway? Not sure about the `profiling data emitted as log records`, but the client-side is a larger vertical and so every instrumentation scope emitted from the client-side will have it repeated. I guess this will require a different example to illustrate the point.

Reply:
If I have an app from which I want to emit profiling data, client side events, and regular logs all represented using OTLP LogRecord, then I need to somehow different the 3 kind of LogRecord although they are all coming from the same Resource (same app).

Reply:
Logs and Events, when both represented using LogRecord, will be distinguishable using the event.name attribute that exists only for events. So this idea does seem only like a mechanism for compression and avoiding repetition of attributes on each signal. If that's the intention, I am wondering why we don't want to take it one level higher above InstrumentationScope, maybe, by introducing another scope layer above InstrumentationScope if we want to have these attributes in resource. 

As an example, the `page.url` attribute representing the base web page will have to be present in all signals emitted while the end-user is on a certain page, however it is not going to be in resource as it doesn't identify the resource. So it is a good candidate to be a top-level scope for all signals, across all InstrumentationScopes.

Reply:
>I am wondering why we don't want to take it one level higher above InstrumentationScope, maybe, by introducing another scope layer above InstrumentationScope if we want to have these attributes in resource.

We briefly chatted in Log SIG about this today, posting for others:
- If you mean adding another layer between Resource and Scope then that is hardly doable since it is unclear how to represent this in OTLP. It would be a major breaking change for OTLP which we cannot allow.
- If you mean actually having these attributes in the resource then I am not sure I understand. We already have resource attributes, why would we have another set?

Reply:
Either way, it seems like this is outside the scope of this particular OTEP. It can be discussed separately and become a new OTEP.